CONTRIBUTING.md

```diff
@@ -391,3 +392,53 @@ Here is an example of such lightweight workflow in Linux environment (Debian-bas
 Maintainer: [@stefanseefeld](https://github.com/stefanseefeld)
 
 _TODO_
+
+## Guidelines
+
+Boost.GIL is a more than a decade old mature library maintained by several
+developers with help from a couple of dozens contributors.
+It is important to maintain consistent design, look and feel.
+Thus, below a few basic guidelines are listed.
+
+First and foremost, make sure you are familiar with the official
+[Boost Library Requirements and Guidelines](https://www.boost.org/development/requirements.html).
+
+Second, strive for writing idiomatic C++11, clean and elegant code.
+
+**NOTE:** *The Boost.GIL source code does not necessary represent clean and elegant
+code to look up to. The library has recently entered the transition to C++11.
+Major refactoring overhaul is ongoing.*
+
+Maintain structure your source code files according to the following guidelines:
+
+* Name files in meaningful way.
+* Put copyright and license information in every file
+* If your changes [meet a certain threshold of originality](https://www.boost.org/users/license.html),
+  add yourself to the copyright notice. Do not put any additional authorship or
+  file comments (eg. no `\file` for Doxygen), revision information, etc.
+* In header, put `#include` guard based on header path and file name
+    ```
+    #ifndef BOOST_GIL_<DIR1>_<DIR2>_<FILE>_HPP
+    #define BOOST_GIL_<DIR1>_<DIR2>_<FILE>_HPP
+    ...
+    #endif
+    ```
+* Group `#include` directives and order alphabetically within each group, as follows:
+    * In headers of GIL core and extensions:
+      1. `#include <boost/gil/extension/*.hpp>`
+      2. `#include <boost/gil/*.hpp>`
+      3. `#include <boost/*.hpp>`
+      4. C++ standard library headers
+      5. Other third-party and system headers
+    * In programs for tests and examples:
+      1. `#include <boost/gil/*.hpp>`
+      2. `#include <boost/*.hpp>`
+      3. C++ standard library headers
+      4. Other third-party and system headers
+      5. `#include "xxx.hpp"` for directory local headers
+* All public headers should be placed in `boost/gil/` or `boost/gil/<component>/`.
```

Comment:
This (as well as the next lines) is good !

Reply:
So, to get it right, you suggest to remove from
`* Group ...`
to
`     5. `#include "xxx.hpp"` for directory local headers`
inclusive. Correct?

Reply:
Yes. (I'm not against the idea to order includes. I'm against putting such detail into a coding guide.)
However, you make a good point: it might be good to add a point:
* make sure that headers are self-contained, i.e. that they include all headers they need.

Reply:
Yes ! :-)

Reply:
OK

Reply:
Done, but with a tweest ;)

I've moved the detailed guidelines on the include directives as "here is what can help" Wiki:
https://github.com/boostorg/gil/wiki/Include-Directives-Order

I think it's quite helpful to have this order listed somewhere. I hope it's OK now.

Reply:
Perfect, thanks !